### PR TITLE
[global] Using lib-helm instead helm_lib directory

### PR DESCRIPTION
**Pull request by name212**
## Description
Using [lib-helm](https://github.com/deckhouse/lib-helm) instead helm_lib directory

## Why do we need it, and what problem does it solve?
3rd-party modules enhancement. `lib-helm` should use for build-in deckhouse modules and for 3rd-party modules. 

## What is the expected result?
Everything works.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: deckhouse-controller
type: feature
summary: Using lib-helm instead helm_lib directory
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/openstack'...

**Comment by name212** [comment]
/e2e/abort openstack;Standard;containerd;1.23 4082153409 4082153409-1-con-1-23 /sys/deckhouse-oss/install:pr3665 debian@46.161.52.109

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/openstack'...

**Comment by name212** [comment]
/e2e/abort openstack;Standard;containerd;1.23 4082473259 4082473259-1-con-1-23 /sys/deckhouse-oss/install:pr3665 debian@46.161.52.109

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/openstack'...

File: Makefile [hunk 1]
```diff
@@ -244,3 +244,8 @@ bin/werf: bin bin/trdl ## Install werf for images-tags generator.
 .PHONY: update-k8s-patch-versions
 update-k8s-patch-versions: ## Run update-patchversion script to generate new version_map.yml.
 	cd candi/tools; bash update_kubernetes_patchversions.sh
+
+##@ Lib helm
+.PHONY: update-lib-helm
+update-lib-helm:
+	cd helm_helpers/ && yq -i '.dependencies[0].version = "$(version)"' Chart.yaml && helm dependency update && tar -xf charts/deckhouse_lib_helm-*.tgz -C charts/ && git add Chart.yaml Chart.lock charts/*
```
Comment (shvgn) [inline Makefile:251]: The archive file is not being cleaned. Do we need it committed? I believe we should add gitignore entry for it

> charts/deckhouse_lib_helm-*.tgz

File: helm_helpers/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v2
+type: library
+name: helm_helpers
```
Comment (shvgn) [inline helm_helpers/Chart.yaml:3]: Why we use new name here for the same purpose? Why not "helm_lib"? The directory "helm_lib" could be preserved.

**Review by shvgn — CHANGES_REQUESTED** [review]

**Review by RomanenkoDenys — COMMENTED** [review]
some nits

File: Makefile (hunk 1 above)
Comment (name212) [reply Makefile:251]: added to gitignore

File: helm_helpers/Chart.yaml (hunk 2 above)
Comment (name212) [reply helm_helpers/Chart.yaml:3]: Yuriy and I decided to name the directory helm_helpers, as he becomes a chart and uses lib-helm as a dependency.

**Comment by name212** [comment]
@RomanenkoDenys It seemed to me and Yuri that it would be more correct to use helm dependency, since it is vanilla, so we didn’t even consider the path with git submodules.

**Review by RomanenkoDenys — APPROVED** [review]

File: helm_helpers/Chart.yaml (hunk 2 above)
Comment (shvgn) [reply helm_helpers/Chart.yaml:3]: Well, it's definitely [multiplying entities beyond necessity](https://en.wikipedia.org/wiki/Occam%27s_razor).

I think "helm_lib" chart in Deckhouse and "deckhouse_lib" chart as it's dependency should be distinguished. In Deckhouse code, helm lib is a just a lib, as always, regardless the dependencies it is using. Like "python_lib" is called "deckhouse" in pip, while it is still in the repo deckhouse/python_lib and in Deckhouse, we have python_lib. 

It's just different contexts. Why shouldn't they be accounted? 

So, I'd not agree on that.

Comment (name212) [reply helm_helpers/Chart.yaml]: fixed. revert helm_lib directory

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/openstack'...

**Review by yalosev — APPROVED** [review]

**Review by shvgn — APPROVED** [review]

**Pull request merged**